xbmc/FileItem.cpp

```diff
@@ -1270,7 +1270,7 @@ bool CFileItem::IsAndroidApp() const
 
 bool CFileItem::IsStack() const
 {
-  return URIUtils::IsStack(m_strPath);
+  return URIUtils::IsStack(GetDynPath());
```

Comment:
Just saying that all (?) `CFileItem::Is`methods working on `m_strPath`. With this change this method now is the exception. Not saying this is wrong, but looks like a big inconsistency.

Reply:
I think in the end, if we think hard enough about obscure use-cases the others may be in fact wrong. It's not by chance that all these misbehaviours happen we we start moving constant paths to playlists (e.g. strm files) or paths are served from plugins which need resolution.
I mean it may cause regressions but from a logical POV IsStack applied to a CFileItem should look always first to the dynpath (may have been resolved) and only after that to the path (signature of the item). If that is not the intended behaviour the client code should invoke `URIUtils::IsStack` directly with the path it wishes to check.
I think the comment may not be necessary here

Reply:
> if we think hard enough about obscure use-cases the others may be in fact wrong.

I tend to agree.